lib/active_merchant/billing/gateways/card_connect.rb

```diff
@@ -63,16 +63,9 @@ class CardConnectGateway < Gateway
 
       def initialize(options = {})
         requires!(options, :merchant_id, :username, :password)
-        require_valid_domain!(options, :domain)
         super
       end
 
-      def require_valid_domain!(options, param)
```

Comment:
This looks to me like it both checks for a port and enforces that the URL contains `cardconnect.com`. It looks to me like the card only wants us to not check for the inclusion of a port anymore. Are we sure we want to get rid of both of these checks?

Reply:
Interesting question! My assumption had been that the method was built around requiring a domain, but looking at things more closely, I think that you're right that it's worth continuing to enforce the `cardconnect.com` bit. I've fixed that up now.

Reply:
It looks like this check was specifically included on implementation so I'm in favor of keeping it until specifically told to do otherwise: https://github.com/activemerchant/active_merchant/pull/2706#discussion_r160954583

Reply:
Oooh, nice digging! Thanks for uncovering that! I went ahead and pushed some changes so that this just removes the port portion.